src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/xattention.hpp

```diff
@@ -0,0 +1,429 @@
+        _kv_head_groups = H / K_H;
+        // The k length should first divided by stride, and the result should be divisible by 32 since block
+        // size in brgemm computation is 32. Therefore align k to multiple of xattn_stride * 32.
+        auto k_padded = rnd_up(B, xattn_stride * 32);
+        _k_num_to_pad = k_padded - B;
+        _k_num_strided = div_up(k_padded, xattn_stride);
+
+        _q_num_strided = div_up(B, xattn_stride);
+        _q_num_blocks = div_up(B, xattn_block_size);
+        _k_num_blocks = div_up(B, xattn_block_size);
+        _num_per_block = xattn_block_size / xattn_stride;
+        size_t n_num_blocks = _k_num_strided / _n_block_size;
+
+        if (_q_num_blocks <= 1 || _k_num_blocks <= 1 || _num_per_block == 0) {
+            return;
+        }
+
+        if (_xattn_gemm.empty()) {
+            _xattn_gemm.resize(_m_block_size);
+            for (size_t i = 1; i < _m_block_size + 1; i++) {
+                _xattn_gemm[i - 1] = std::make_shared<BrgemmKernel>(i,                         // M
+                                                                    _n_block_size,             // N
+                                                                    S,                         // K
+                                                                    S * xattn_stride * H * L,  // lda
+                                                                    _n_block_size,             // ldb
+                                                                    _k_num_strided,            // ldc
+                                                                    false,
+                                                                    in_type,
+                                                                    true);
+            }
+            auto max_threads_num = parallel_get_max_threads();
+            auto scratch_a_size = _xattn_gemm.back()->get_scratch_a_size();
+            auto wsp_size = _xattn_gemm.back()->get_wsp_size();
+
+            _scratch_a.assign(scratch_a_size * max_threads_num, 0.0f);
+            _wsp.assign(wsp_size * max_threads_num, 0.0f);
+
+            _attn_sum_temp.resize({H, L, _q_num_strided, _k_num_strided}, sizeof(float), ov::element::Type_t::f32);
+            ov::parallel_for3d(_q_num_strided, H, L, [&](size_t b, size_t h, size_t l) {
+                memset(_attn_sum_temp.ptr<float>(h, l, b, 0), 0, _k_num_strided * sizeof(float));
+            });
+
+            _attn_sum.resize({H, L, _q_num_blocks, _k_num_blocks}, _attn_sum_temp.m_element_size, _attn_sum_temp.m_dt);
+
+            _key_repack.resize({K_H, L, S * n_num_blocks, _n_block_size}, sizeof(float), ov::element::Type_t::f32);
+        }
+    }
+
+    void estimate(const PlainTensor& query,
+                  const PlainTensor& key,
+                  size_t block_size,
+                  size_t stride,
+                  float threshold,
+                  PlainTensor& mask) {
+        auto B = query.size(0);
```

Comment:
Please add const to constant data in the following
```suggestion
        const auto B = query.size(0);
```